### PR TITLE
update jdk 11 in dockerfile and remove unused java_opts.

### PR DESCRIPTION
**Pull request by JaredTan95**
as https://github.com/apache/skywalking-docker/pull/22#discussion_r681776376 mentioned.
- [ ] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/CHANGES.md).

### CODE REVIEW

File: docker/agent/Dockerfile.agent [hunk 1]
```diff
@@ -14,7 +14,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-ARG BASE_IMAGE='adoptopenjdk/openjdk8:alpine'
+ARG BASE_IMAGE='adoptopenjdk/openjdk11:alpine'
```
Comment (wu-sheng) [inline docker/agent/Dockerfile.agent:17]: Do you upgrade the agent's JDK? I am good for moving OAP to JDK11.

**Review by kezhenxu94 — DISMISSED** [review]

File: docker/agent/Dockerfile.agent (hunk 1 above)
Comment (kezhenxu94) [reply docker/agent/Dockerfile.agent:17]: > Do you upgrade the agent's JDK? I am good for moving OAP to JDK11.

This is only a default value and can be overrode, it's good to me, we will finally release multiple Docker images based on different JDK version

Comment (JaredTan95) [reply docker/agent/Dockerfile.agent:17]: agent image's JDK is not used, right?  except users use agent image as their java application runtime base image. @kezhenxu94

Comment (kezhenxu94) [reply docker/agent/Dockerfile.agent:17]: > agent image's JDK is not used, right? except users use agent image as their java application runtime base image. @kezhenxu94

Yes right

Comment (wu-sheng) [reply docker/agent/Dockerfile.agent:17]: I am curious, where do we use this agent image? If not, why upgrade? Do we plan to use it somewhere?
JDK8 is the version we announced the minimal supported JVM version, and still in maintain mode by OpenJDK and vendors. Why are we going rid of it now?

Comment (JaredTan95) [reply docker/agent/Dockerfile.agent:17]: > JDK8 is the version we announced the minimal supported JVM version

If we have this consideration, then we don't need to update it, I will revert this change.

Comment (wu-sheng) [reply docker/agent/Dockerfile.agent]: Thanks.

**Review by wu-sheng — APPROVED** [review]

**Review by hanahmily — APPROVED** [review]
LGTM.
BTW, Could you use a released CLI to replace the current building from scratch? It will reduce the building time a lot.

**Comment by wu-sheng** [comment]
Please check why CI fails.

**Comment by JaredTan95** [comment]
> Please check why CI fails.

I re-run failed jobs.

**Pull request merged**